runtime/compiler/runtime/SymbolValidationManager.cpp

```diff
@@ -81,6 +79,12 @@ TR::SymbolValidationManager::defineGuaranteedID(void *symbol, TR::SymbolType typ
    _seenSymbolsSet.insert(symbol);
    }
 
+bool
+TR::SymbolValidationManager::isDefinedID(uint16_t id)
```

Comment:
Should some of the existing `isAlreadyValidated` / `tryGetIDFromSymbol` be replaced with this (faster) query?

Reply:
`isAlreadyValidated` takes a pointer in order to check during compilation that we've already established how loads will find the corresponding pointer later. OTOH `isDefinedID` takes an ID and tells us whether the value for that ID has been determined at load time